File: android/app/src/main/java/com/onurvar/massiverpg/domain/usecase/SignOutUseCaseImpl.kt
package com.onurvar.massiverpg.domain.usecase

import com.onurvar.massiverpg.domain.protocol.repository.AuthRepository
import com.onurvar.massiverpg.domain.protocol.usecase.SignOutUseCase
import javax.inject.Inject

class SignOutUseCaseImpl @Inject constructor(
    private val authRepository: AuthRepository
) : SignOutUseCase {
    override suspend fun execute() {
        authRepository.signOut()
    }
}

File: ios/MassiveRPG/Domain/UseCase/SignOutUseCaseImpl.swift
//
//  SignOutUseCaseImpl.swift
//  MassiveRPG-Prod
//
//  Created by Onur Var on 9.04.2024.
//

class SignOutUseCaseImpl: SignOutUseCase {
    // MARK: Variables

    let authRepository: AuthRepository

    // MARK: Life Cycle

    init(authRepository: AuthRepository) {
        self.authRepository = authRepository
    }

    // MARK: Methods

    func execute() throws {
        try authRepository.signOut()
    }
}
